### DISCUSSION TITLE
Will virtual threads be supported in the future?

### DISCUSSION THREAD

**Discussion by tcmot** [apache/logging-log4j2#2596] [General]
I know Log4j2 uses LMAX Disruptor.
But LMAX Disruptor may never support virtual threads.

**Comment by vy** [comment]
@tcmot, Log4j uses LMAX Disruptor **iff** asynchronous loggers are enabled – they are optional, and disabled by default. Hence, **Log4j supports virtual threads**.

**Comment by tcmot** [comment]
@vy Thank you，Which version? 2.23.1？

**Reply by vy** [reply]
@tcmot, asynchronous loggers have *always* been an opt-in feature.

**Reply by zekronium** [reply]
That is only partly true. Current versions still use synchronized blocks that can cause carrier thread pinning, even without async logging

**Reply by vy** [reply]
@zekronium, please note that Loom will eventually address pinning of `synchronized` – see [this Java 23 EA announcement](https://inside.java/2024/06/22/quality-heads-up/). Hence, I doubt your argument will apply for Java 23 and onwards.

**Reply by zekronium** [reply]
But it will be valid for quite long until next LTS, so quite relevant for the near future.

I also hope that Log4j3, which addresses this issue, will be released sooner than next LTS
